nexus/db-queries/src/db/datastore/external_ip.rs

```diff
@@ -430,6 +433,49 @@ impl DataStore {
         self.allocate_external_ip(opctx, data).await
     }
 
+    /// List one page of all external IPs allocated to internal services
+    pub async fn service_ip_all_list(
```

Comment:
I read this as a reversal of the usual convention of `service_ip_list_all` which would mean "list all the service IPs" but I guess it's really "list _a page_ of "all" service IPs"?  What's the "all" here?

Reply:
There's already a `service_lookup_external_ips` which lists the IPs for a specific service. The "all" here is supposed to be "external IPs for _all_ services". I renamed this function 2 or 3 times and never landed on something that really felt right - any suggestions?

Reply:
I think from scratch I might choose `external_ip_list_all_services` and `external_ip_list_service` (for the one particular service).

Is this the "service" that we now call "zone"?  Maybe it would be `external_ip_list_all_zones` and `external_ip_list_zone`?

Whatever you think.  I also think it's fine as is.

Reply:
`external_ip_list_all_service{,s}` sounds good to me. I'm less sure about calling it "zone"; external IPs can also be attached to instances, which also have zones. All the networking-related datastore bits try to label things as "service" or "instance".

Reply:
> All the networking-related datastore bits try to label things as "service" or "instance".

Definitely outside the scope of this PR but I wonder if we want to revisit this, at least for "service".  I would guess this code uses "service" because that's what we used to call them.  We moved away from that term elsewhere.  RFD 248 defines "service" the way I think the industry usually uses that term, which is in terms of how the _consumer_ views it.  i.e., one "network service" might be a DNS name with many IPs corresponding to a potentially different number of backends.  It's super confusing to call each of those backends a separate "service".  There's also the confusion with SMF services, though SMF services themselves match the more consumer-oriented view (e.g., in having services like "filesystem/local" which means "persistent local filesystems are available", even though that's not necessarily a discrete OS process).

It's doubly unfortunate that broadly in the industry, "instance" means both "a virtual machine" and "one of many [basically fungible] backends for a service".  It'd be really nice if we had a notation or a more specific term we could use.